### PR TITLE
Enable golangci-lint stylecheck linter, fix errors

### PR DESCRIPTION
**Pull request by dfarrell07**
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ linters:
     # - sqlclosecheck # We don't use SQL
     - staticcheck
     - structcheck
-    # - stylecheck # TODO: Fix and enable
+    - stylecheck # TODO: Fix and enable
```
Comment (tpantelis) [inline .golangci.yml:92]: ?

```suggestion
    - stylecheck
```

File: .golangci.yml [hunk 2]
```diff
@@ -142,3 +142,13 @@ issues:
     - linters:
         - lll
       path: "pkg/apis/submarinerconfig/v1alpha1/zz_generated.swagger_doc_generated.go"
+
+    # Ignore underscores in var names in auto-generated files
+    - linters:
+        - stylecheck
+      path: "pkg/apis/submarinerconfig/v1alpha1/zz_generated.swagger_doc_generated.go"
+
+    # This needs to be changed in the main submariner-io/submariner repo first
```
Comment (tpantelis) [inline .golangci.yml:151]: This is defined in _pkg/apis/submarinerconfig/v1alpha1/types.go_ in this repo and not submariner. I don't think the Go struct is used outside of this repo so should be safe to change it and remove this exclusion.

File: pkg/apis/submarinerconfig/v1alpha1/types.go [hunk 3]
```diff
@@ -176,9 +176,9 @@ type ManagedClusterInfo struct {
 	// Region represents the cloud region of the managed cluster.
 	// +optional
 	Region string `json:"region,omitempty"`
-	// InfraId represents the infrastructure id of the managed cluster.
+	// InfraID represents the infrastructure id of the managed cluster.
 	// +optional
-	InfraId string `json:"infraId,omitempty"`
+	InfraID string `json:"infraID,omitempty"`
```
Comment (tpantelis) [inline pkg/apis/submarinerconfig/v1alpha1/types.go:181]: We shouldn't change rendered representation, just the Go field.

```suggestion
	InfraID string `json:"infraId,omitempty"`
```

File: pkg/apis/submarinerconfig/v1alpha1/zz_generated.swagger_doc_generated.go [hunk 4]
```diff
@@ -33,7 +33,7 @@ var map_ManagedClusterInfo = map[string]string{
 	"vendor":      "Vendor represents the kubernetes vendor of the managed cluster.",
 	"platform":    "Platform represents the cloud provider of the managed cluster.",
 	"region":      "Region represents the cloud region of the managed cluster.",
-	"infraId":     "InfraId represents the infrastructure id of the managed cluster.",
+	"infraID":     "InfraID represents the infrastructure ID of the managed cluster.",
```
Comment (tpantelis) [inline pkg/apis/submarinerconfig/v1alpha1/zz_generated.swagger_doc_generated.go:36]: This shouldn't change.

File: deploy/config/crds/submarineraddon.open-cluster-management.io_submarinerconfigs.yaml [hunk 5]
```diff
@@ -162,8 +162,8 @@ spec:
                   clusterName:
                     description: ClusterName represents the name of the managed cluster.
                     type: string
-                  infraId:
-                    description: InfraId represents the infrastructure id of the managed cluster.
+                  infraID:
```
Comment (tpantelis) [inline deploy/config/crds/submarineraddon.open-cluster-management.io_submarinerconfigs.yaml:165]: The rendered YAML representation shouldn't change.

File: deploy/olm-catalog/0.3.0/submarineraddon.open-cluster-management.io_submarinerconfigs.yaml [hunk 6]
```diff
@@ -131,8 +131,8 @@ spec:
                 clusterName:
                   description: ClusterName represents the name of the managed cluster.
                   type: string
-                infraId:
-                  description: InfraId represents the infrastructure id of the managed cluster.
+                infraID:
+                  description: InfraID represents the infrastructure ID of the managed cluster.
```
Comment (tpantelis) [inline deploy/olm-catalog/0.3.0/submarineraddon.open-cluster-management.io_submarinerconfigs.yaml:135]: We can't change the YAML representation.

```suggestion
                infraId:
                  description: InfraId represents the infrastructure ID of the managed cluster.
```

File: deploy/olm-catalog/manifests/submarineraddon.open-cluster-management.io_submarinerconfigs.yaml [hunk 7]
```diff
@@ -160,8 +160,8 @@ spec:
                   clusterName:
                     description: ClusterName represents the name of the managed cluster.
                     type: string
-                  infraId:
-                    description: InfraId represents the infrastructure id of the managed cluster.
+                  infraID:
+                    description: InfraID represents the infrastructure ID of the managed cluster.
```
Comment (tpantelis) [inline deploy/olm-catalog/manifests/submarineraddon.open-cluster-management.io_submarinerconfigs.yaml:164]: Same here.

```suggestion
                  infraId:
                    description: InfraId represents the infrastructure ID of the managed cluster.
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/apis/submarinerconfig/v1alpha1/types.go [hunk 8]
```diff
@@ -178,7 +178,7 @@ type ManagedClusterInfo struct {
 	Region string `json:"region,omitempty"`
 	// InfraId represents the infrastructure id of the managed cluster.
 	// +optional
-	InfraId string `json:"infraId,omitempty"`
+	InfraID string `json:"infraId,omitempty"`
```
Comment (Jaanki) [inline pkg/apis/submarinerconfig/v1alpha1/types.go:181]: This also can't be changed IMO. It would mean makjng API changes and non-backward compatibility,

Comment (tpantelis) [reply pkg/apis/submarinerconfig/v1alpha1/types.go:181]: It's OK to make Go API changes b/c it's an opt-in for users of the code to pull it in, ie they have to update their version in the go.mod and then adjust their code accordingly. However I don't think anything outside this repo is using this code anyway.

**Pull request merged**